jstz_cli/src/run.rs

```diff
@@ -12,23 +11,53 @@ use crate::{
 };
 
 pub fn exec(
-    cfg: &Config,
-    referrer: String,
+    cfg: &mut Config,
+    referrer: Option<String>,
     url: String,
     http_method: String,
     json_data: Option<String>,
 ) -> Result<()> {
-    // Create transaction
-    let tx = ExternalMessage::Transaction(Transaction {
-        referrer: PublicKeyHash::from_base58(&referrer).unwrap(),
-        url,
-        method: Method::from_str(&http_method).unwrap(),
-        body: json_data.map(from_file_or_id).or_else(piped_input),
-    });
+    let account = cfg.accounts.account_or_current_mut(referrer)?;
 
-    // Create JSON message
-    let jmsg = serde_json::to_string(&tx).unwrap();
+    // Create operation TODO nonce
+    let url: Uri = url
+        .parse()
+        .map_err(|_| anyhow!("Failed to parse URL: {}", url))?;
+
+    let method =
+        Method::from_str(&http_method).map_err(|_| anyhow!("Invalid HTTP method"))?;
+
+    let body = json_data
+        .map(from_file_or_id)
+        .or_else(piped_input)
+        .map(String::into_bytes);
+
+    let op = Operation {
+        source: account.address.clone(),
+        nonce: account.nonce.clone(),
+        content: Content::RunContract(RunContract {
+            uri: url,
+            method,
+            headers: HeaderMap::default(),
+            body,
+        }),
+    };
+
+    account.nonce.increment();
+
+    let signed_op = SignedOperation::new(
+        account.public_key.clone(),
+        account.secret_key.sign(op.hash())?,
+        op,
+    );
+
+    let json_string = serde_json::to_string_pretty(&serde_json::to_value(&signed_op)?)?;
```

Comment:
Nit:
I think it would be better to serialise twice here. A pretty printed version for the debug long and a non-pretty version for the rollup.